src/app/page.tsx

```diff
@@ -1,8 +1,16 @@
 // this seems like an open issue, search params returned empty object, so had to force-dynamic : https://github.com/vercel/next.js/issues/43077
-export const dynamic='force-dynamic';
+export const dynamic = "force-dynamic";
 
 import ClientPage from "./clientPage";
+import About from "~/components/About";
+import { isValidUrlWithEncodedState } from "~/utils/isValidUrlWithEncodedState";
 
 export default function Page({ searchParams }: { searchParams?: { [key: string]: string | string[] | undefined } }) {
-  return <ClientPage searchParams={searchParams} />;
+  return (
+    <>
+      <ClientPage searchParams={searchParams} />
+      {/* Hide about page in the results page */}
+      {!isValidUrlWithEncodedState(searchParams) && <About />} 
```

Comment:
This check isn't working too well for me. Almost like the state is not reactive

Reply:
do you know if there is a better way to check if the parsed string will be a JSON? I think this test would fail if the query params value are numbers, something like this /?result=123&original=2323...

I have tested the following cases:

[FAIL:  it goes to the Result page but is blank // it should have stayed in the homepage with about section displayed]
https://deploy-preview-26--sum-it-up.netlify.app/?original=2323&result=123

[PASS: it stays on the homepage as intended and shows About section as intended]
https://deploy-preview-26--sum-it-up.netlify.app/?original=&result=  
https://deploy-preview-26--sum-it-up.netlify.app/?original=abc&result=abc

Reply:
I will do some more tests and fix this later.. also I think we could even show the About section in the Result page, (we wouldn't really need to check the state then 😂)

Reply:
Ok I see the state also getting messed up after returning to the main page from the results page, will fix this later

Reply:
as for this question...
>do you know if there is a better way to check if the parsed string will be a JSON? I think this test would fail if the query params value are numbers, something like this /?result=123&original=2323...

Didn't even notice that, nice catch. Perhaps we could check if the result has properties.